### ISSUE TITLE
"No matching distribution found for tensorflow" when installing with pip on macOS 

### ISSUE THREAD

**Issue by KathyGCY** [tensorflow/tensorflow#60512]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.3.0

### Custom Code

Yes

### Python version

3.9

### Current Behaviour?

`ERROR: No matching distribution found for tensorflow==2.3.0`

### Standalone code to reproduce the issue

```shell
pip install -U --index-url https://pypi.org/ tensorflow==2.3.0
```

### Relevant log output

```shell
Same issue as here: "No matching distribution found for tensorflow" when installing with pip on macOS 11 #47205, https://github.com/tensorflow/tensorflow/issues/47205
The issue did not resolve but was closed.
The solution doesn't work it errored with "ERROR: No matching distribution found for tensorflow==2.3.0"
```
</details>

**Comment by sampathweb** [comment]
@KathyGCY - TensorFlow 2.3 is 2+ years old (July'2000) and was released for Python 3.5-3.8 (https://pypi.org/project/tensorflow/2.3.0/#files).  You have a Python 3.9 Env.  Please use a more recent TensorFlow version, preferably TensorFlow 2.12 (latest).

**Comment by sampathweb** [comment]
Going to close this issue.  Please re-open if you have problem with installing newer version of TensorFlow in your Python 3.9 env.  Follow steps in https://www.tensorflow.org/install/pip#macos
If you have a Mac M1/M2, there are instructions in the installs page that's slightly different.
